### PR TITLE
[ROCm] Enable sort operator BF16 support

### PR DESCRIPTION
**Pull request by pruthvistony**
The changes add support for dtype BF16 for sort operator in ROCm.

Relates - https://github.com/pytorch/pytorch/pull/58196

Relanding the change - https://github.com/pytorch/pytorch/pull/71226

@jeffdaily @jithunnair-amd @dllehr-amd Please review this PR.

### CODE REVIEW

**Comment by pruthvistony** [comment]
Copied Jithun's comments

@pruthvistony @dllehr-amd Can you please elaborate on the breakage? The argsort unit test was enabled for float16 on ROCm by this PR: https://github.com/pytorch/pytorch/pull/67706/files#diff-18a0c3e7cd07089b34760028c387d39de1aa393172f1d5f123692106b3a3d144R15390

but CI passed for that PR. And I'm not sure how PR https://github.com/pytorch/pytorch/pull/71226 impacted it since it was for bfloat16?

--- There are 2 possible solutions add BF16 to the argsort ROCm dtype or removing the dtype specialization for ROCm.

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -13285,7 +13285,7 @@ def ref_pairwise_distance(input1, input2):
     OpInfo('sort',
            dtypes=all_types_and(torch.bool, torch.float16, torch.bfloat16),
            dtypesIfCUDA=all_types_and(torch.float16, torch.bfloat16),
-           dtypesIfROCM=all_types_and(torch.float16),
+           dtypesIfROCM=all_types_and(torch.float16, torch.bfloat16),
```
Comment (dllehr-amd) [inline torch/testing/_internal/common_methods_invocations.py:13288]: Like the argsort, we can probably remove `dtypesIfROCM` from here and in `msort`

**Review by dllehr-amd — CHANGES_REQUESTED** [review]
Looks good minus a few comments with `dtypesIfROCM`

File: torch/testing/_internal/common_methods_invocations.py (hunk 1 above)
Comment (pruthvistony) [reply torch/testing/_internal/common_methods_invocations.py:13288]: agree, better to keep in uniform

File: aten/src/ATen/cuda/cub.cu [hunk 2]
```diff
@@ -57,10 +57,7 @@ AT_INSTANTIATE_SORT_PAIRS(int64_t, 4)
 
 AT_FORALL_SCALAR_TYPES_AND2(Bool, Half, AT_INSTANTIATE_SORT_PAIRS_8)
 
-// BFloat16 is not supported by ROCm's radix sort
-#if !AT_ROCM_ENABLED()
```
[the remark this answers is not in the document]
Comment (pruthvistony) [reply aten/src/ATen/cuda/cub.cu:61]: added the check.

**Review by pruthvistony — COMMENTED** [review]
Changes are updated.

**Review by pruthvistony — COMMENTED** [review]
Review comments are taken care.

**Review by jithunnair-amd — APPROVED** [review]

**Review by dllehr-amd — APPROVED** [review]
LGTM!

**Review by malfet — APPROVED** [review]

**Pull request closed without merging**